Orleans/src/Microsoft.Orleans.Host.Abstractions/StorageProvidersFeature.cs

```diff
@@ -0,0 +1,50 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the (optional) storage providers DLL.
+    public interface IStorageProvidersFactory
+    {
+        ISiloBuilder Silo { get; }
```

Comment:
Because it allows you to access runtime service that can be used by it or passed along for concrete provider factories to use, as in this azure storage example